### PR TITLE
calculate focal of camera and distance from object

### PR DESCRIPTION
**Pull request by Mboshy**

### CODE REVIEW

**Review by DominikKwiatkowski — CHANGES_REQUESTED** [review]
please use pre-commit hooks from tutorial and fix all coding style issues

**Comment by Mboshy** [comment]
I need to investigate what is wrong. It passed all tests when I commited via git bash, but on github few failed

File: drones/image_processing/__init__.py [hunk 1]
```diff
@@ -1 +1,2 @@
 import drones.image_processing.normalization as normalization
+from drones.image_processing.utils import calculate_focal, distance_to_camera
```
Comment (Mboshy) [inline drones/image_processing/__init__.py]: I have no idea at the moment how to solve this problem with "No newline at end of file"

**Comment by Mboshy** [comment]
Does anyone know what should I fix? I see that there is an issue with __init__.py. No clue what is wrong

**Comment by bazylip** [comment]
All files should end in a newline. Add newline at the end of `__init__.py`. Black should reformat it on its own, so there is a high chance your pre-commit setup is wrong. Flake8 will probably also fail, because your variable names are not PEP8 compliant.

File: drones/image_processing/utils.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    focal_length: float
+        calculated length of the focal
+
+    Raises:
+    ----------
+        ZeroDivisionError: when nominator (known_width) is equal 0
+    """
+    try:
+        focal_length = (pixel_width * known_distance) / known_width
+    except ZeroDivisionError as err:
```
Comment (bazylip) [inline drones/image_processing/utils.py]: There is no need to catch an exception just to raise it in the following line.

File: drones/image_processing/utils.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    distance: float
+        Calculated distance between detected object and camera
+
+    Raises:
+    ----------
+        ZeroDivisionError: when nominator (pixel_width) is equal 0
+    """
+    try:
+        distance = (known_width * focal_length) / pixel_width
+    except ZeroDivisionError as err:
```
Comment (bazylip) [inline drones/image_processing/utils.py]: Same exception issue here.

File: drones/image_processing/utils.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        raise ZeroDivisionError(err)
+
+    return distance
+
+
+def vector_to_centre(frame_width: int, frame_height: int, obj_coordinates: tuple, centre_height_coeff: float) -> tuple:
```
Comment (bazylip) [inline drones/image_processing/utils.py]: Please use `typing.Tuple` in type hints.

**Review by bazylip — CHANGES_REQUESTED** [review]

File: drones/image_processing/utils.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    ----------
+    distance: float
+        Calculated distance between detected object and camera
+    """
+    distance = (known_width * focal_length) / pixel_width
+
+    return distance
+
+
+def vector_to_centre(frame_width: int, frame_height: int, obj_coordinates: Tuple, centre_height_coeff: float) -> Tuple:
```
Comment (bazylip) [inline drones/image_processing/utils.py]: Please specify tuple members' type as well. Take a look [here](https://docs.python.org/3/library/typing.html#typing.Tuple) to see more examples.

**Review by bazylip — CHANGES_REQUESTED** [review]

**Review by DominikKwiatkowski — APPROVED** [review]

**Review by bazylip — APPROVED** [review]

**Pull request merged**